libs/coin-framework/src/batcher/index.ts

```diff
@@ -0,0 +1,65 @@
+import { Batch, BatchElement, Batcher } from "./types";
+
+/**
+ * Factory to make an API call batcher.
```

Comment:
I understand that it is used to kind of do a "Promise.all" between multiple files, it it to "compensate" for some architecture problem of our coin framework ?

Would we have another way to do it ?

Reply:
It's not compensating for the coin-framework, it's a solution like https://github.com/graphql/dataloader. 
Basically, it allows multiple frontend components to ask in their own promise "fetch this param" while on the other end all the params are going to be grouped into a single "batched" request.
(so a `POST /endpoint ([param1, param2, param3]): [response1, response2, response3]`) 
And the responses will be returned as a resolution to each original promise individually. 

```
                                                                           
                    +----------+  +----------+                             
                    |          |  |          |                             
                    |  comp 1  |  |  comp 2  |                             
                    |          |  |          |                             
                    +----------+  +----------+                             
            +--+ promise1|            |promise2 +--+                       
            |    param   |            |  param2    |                       
            |            |            |            |                       
 resolve(A) |            +            +            | resolve(B)            
            |           +--------------+           |                       
            |           |              |           |                       
            +-----------|    batcher   |-----------+                       
                        |              |                                   
                        +--------------+                                   
                     params  |     + returns                               
                     [1,2]   |     | [A,B]                                 
                             +     |                                       
                         +------------+                                    
                         |            |                                    
                         |   server   |                                    
                         |            |                                    
                         +------------+                                    
                                                                           
                                                                           
                                                                           
```